cmd/manager/main.go

```diff
@@ -80,28 +83,21 @@ func main() {
 		os.Exit(1)
 	}
 
-	ctx := context.TODO()
-
-	// Become the leader before proceeding
-	err = leader.Become(ctx, "watermarkpodautoscaler-lock")
-	if err != nil {
-		log.Error(err, "")
-		os.Exit(1)
-	}
-
 	// Create a new Cmd to provide shared dependencies and start components
 	mgr, err := manager.New(cfg, manager.Options{
-		Namespace:          namespace,
-		MapperProvider:     apiutil.NewDiscoveryRESTMapper,
-		MetricsBindAddress: fmt.Sprintf("%s:%d", metricsHost, metricsPort),
+		Namespace:              namespace,
+		MapperProvider:         apiutil.NewDiscoveryRESTMapper,
+		MetricsBindAddress:     fmt.Sprintf("%s:%d", host, metricsPort),
```

Comment:
maybe `net.JoinHostPort` can be used here. but I don't remember if the port is an `int32`

Reply:
That's been working well in my testings, I got some inspiration from other upstream controllers